File: backend/src/main/java/com/decmoe47/todo/model/entity/TodoList.java
package com.decmoe47.todo.model.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.experimental.Accessors;
import org.hibernate.annotations.UuidGenerator;

import java.util.List;

@EqualsAndHashCode(callSuper = true)
@Accessors(chain = true)
@Data
@Entity
public class TodoList extends Auditable<User> {

    @Id
    @GeneratedValue
    @UuidGenerator(style = UuidGenerator.Style.TIME)
    private String id;

    private String name;

    private Boolean isInbox = false;

    @OneToMany(mappedBy = "belongedList")
    private List<Todo> todos;
}

File: backend/internal/model/entity/todo_list.go
package entity

import "gorm.io/gorm"

type TodoList struct {
	*Auditable
	Name  string
	Inbox bool

	Todos []*Todo `gorm:"foreignKey:BelongedListID"`
}

func NewTodoList(name string, createdByID uint, inbox bool) *TodoList {
	return &TodoList{
		Auditable: &Auditable{
			Model:       &gorm.Model{},
			CreatedByID: createdByID,
		},
		Name:  name,
		Inbox: inbox,
	}
}
